sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -961,33 +985,48 @@ public Flux<OpenConnectionResponse> openConnectionsAndInitCaches(
-                            .filter(addressInfo -> this.protocolScheme.equals(addressInfo.getProtocolScheme()))
-                            .collect(Collectors.groupingBy(Address::getParitionKeyRangeId))
-                            .values()
-                            .stream().map(addresses -> toPartitionAddressAndRange(collection.getResourceId(), addresses))
-                            .collect(Collectors.toList());
+                                    .filter(addressInfo -> this.protocolScheme.equals(addressInfo.getProtocolScheme()))
+                                    .collect(Collectors.groupingBy(Address::getParitionKeyRangeId))
+                                    .values()
+                                    .stream().map(addresses -> toPartitionAddressAndRange(collection.getResourceId(), addresses))
+                                    .collect(Collectors.toList());
 
                     return Flux.fromIterable(addressInfos)
-                            .flatMap(
-                                addressInfo -> {
-                                    this.serverPartitionAddressCache.set(addressInfo.getLeft(), addressInfo.getRight());
-
-                                    if (this.openConnectionsHandler != null) {
-                                        return this.openConnectionsHandler.openConnections(
-                                            collection.getResourceId(),
-                                            this.serviceEndpoint,
-                                            Arrays
-                                                .stream(addressInfo.getRight())
-                                                .map(addressInformation -> addressInformation.getPhysicalUri())
-                                                .collect(Collectors.toList()));
-                                    }
-
-                                    logger.info("OpenConnectionHandler is null, can not open connections");
-                                    return Flux.empty();
-                                },
-                                Configs.getCPUCnt() * 10,
-                                Configs.getCPUCnt() * 3);
+                            .flatMap(addressInfo -> {
+                                this.serverPartitionAddressCache.set(addressInfo.getLeft(), addressInfo.getRight());
+                                return Flux.fromArray(addressInfo.getRight());
+                            }, Configs.getCPUCnt() * 10, Configs.getCPUCnt() * 3)
+                            .flatMap(addressInformation -> Mono.just(new ImmutablePair<>(new ImmutablePair<>(containerLink, collection), addressInformation)));
                 });
+
+    }
+
+    public Flux<Void> submitOpenConnectionTask(
+            AddressInformation address,
+            DocumentCollection documentCollection,
+            int connectionsPerEndpointCount
+    ) {
+        if (this.openConnectionsHandler != null && this.proactiveOpenConnectionsProcessor != null) {
```

Comment:
should `proactiveOpenConnectionsProcessor` has `openConnectionsHandler`? what is the benefits of containing a  openConnectionHandler for each openConnectionOperation?